web/src/components/TestCard/TestCard.tsx

```diff
@@ -21,6 +22,7 @@ const TestCard = ({onClick, onDelete, onRunTest, test: {name, serviceUnderTest,
 
   const onCollapse = useCallback(async () => {
     if (resultList.length > 0) {
+      TestAnalyticsService.onTestCardCollapse();
```

Comment:
Maybe this should go outside the if statement. This `if statement` is only to avoid making the API call again if you already collapsed the list.

Reply:
Makes snese

Reply:
But actually, the request is only triggered once either way, as the slice would keep the cached data.

Reply:
Yeah that makes sense. We need to double check that query to verify we are using the redux cache. Maybe not part of this PR.